nautobot_floor_plan/views.py

```diff
@@ -41,6 +41,81 @@ def get_extra_context(self, request, instance=None):
         ]
         context["zoom_duration"] = get_app_settings_or_config("nautobot_floor_plan", "zoom_duration")
         context["highlight_duration"] = get_app_settings_or_config("nautobot_floor_plan", "highlight_duration")
+
+        # Default to showing the default tab
+        context.update(
+            {
+                "x_activate_default_tab": True,
+                "x_activate_custom_tab": False,
+                "y_activate_default_tab": True,
+                "y_activate_custom_tab": False,
+            }
+        )
+
+        # Check for custom labels on the instance first
+        if instance and instance.pk:
+            has_x_custom_labels = instance.custom_labels.filter(axis="X").exists()
+            has_y_custom_labels = instance.custom_labels.filter(axis="Y").exists()
+
+            if has_x_custom_labels:
+                context.update(
+                    {
+                        "x_activate_default_tab": False,
+                        "x_activate_custom_tab": True,
+                    }
+                )
+
+            if has_y_custom_labels:
+                context.update(
+                    {
+                        "y_activate_default_tab": False,
+                        "y_activate_custom_tab": True,
+                    }
+                )
+
+        # Then check form state if available
+        form = context.get("form")
+        if form:
+            # X-axis tab activation logic
+            x_default_tab_errors = (
+                getattr(form, "x_origin_seed", {}).errors
```

Comment:
nit - I dont see this as a PR stopper issue cause you have really solid tests along with it.

just FYI, Pylance is yelling about this default dictionary return because there will be no `.errors` attribute on that. Maybe default to a data_dictionary that has a blank errors attribute?